File: frontend/src/app/models/Speaker.ts
export class Speaker {
  id: number;
  speakerId: string;
  sex: string;
  languageUsed: string;
  dialect: string;

  constructor(id: number, speakerId: string, sex: string, languageUsed: string, dialect: string) {
    this.id = id;
    this.speakerId = speakerId;
    this.sex = sex;
    this.languageUsed = languageUsed;
    this.dialect = dialect;
  }
}

File: backend/src/main/scala/models/Speaker.scala
package models

import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class Speaker(
                    id: String,
                    sex: String,
                    languageUsed: String,
                    dialect: String
                  ) {
}

object Speaker {
  implicit val encoder = deriveEncoder[Speaker]
  implicit val decoder = deriveDecoder[Speaker]
}
